### PR TITLE
672 integrationtests run tests with and without authorization

### PR DESCRIPTION
**Pull request by Ziktorin**

### CODE REVIEW

File: .github/workflows/integration-tests.yml [hunk 1]
```diff
@@ -44,6 +44,11 @@ on:
         description: 'Upper boundry for chain test'
         required: true
         default: '100000'
+      bob_min_port:
```
Comment (ikopylov) [inline .github/workflows/integration-tests.yml:47]: I think there is no need to ask the user to set this parameter, since it does not affect the test in significant way

File: .github/workflows/integration-tests.yml [hunk 2]
```diff
@@ -52,7 +57,7 @@ jobs:
         key: [8, 16]
     env:
       TARGET: "x86_64-unknown-linux-musl"
-      BUILD_MODE: "release"
+      BUILD_MODE: "debug"
```
Comment (ikopylov) [inline .github/workflows/integration-tests.yml:60]: Revert this change in this request. This change is already made in https://github.com/qoollo/bob/pull/524

File: .github/workflows/integration-tests.yml [hunk 3]
```diff
@@ -72,7 +77,7 @@ jobs:
         uses: actions-rs/cargo@v1.0.1
         with:
           command: build
-          args: --release --target=${{ env.TARGET }}
+          args: --target=${{ env.TARGET }}
```
Comment (ikopylov) [inline .github/workflows/integration-tests.yml:80]: Same here

File: .github/workflows/integration-tests.yml [hunk 4]
```diff
@@ -182,17 +192,18 @@ jobs:
       - name: List availible images
         run: |
           docker image ls
+      #Also sets BOB_BOBP_ACTIONS_OFFSET as GITHUB_ENV from check_interval parameter (check_interval + 1s)
```
Comment (ikopylov) [inline .github/workflows/integration-tests.yml:195]: I added comment for this in https://github.com/qoollo/bob/pull/524. In this request that line should be removed

File: .github/workflows/integration-tests.yml [hunk 5]
```diff
@@ -201,6 +212,7 @@ jobs:
     strategy:
       matrix:
         key: [8, 16]
+        auth: [None, Basic]
```
Comment (ikopylov) [inline .github/workflows/integration-tests.yml:215]: There is no need to run aliens test with auth. gRPC auth was already tested within put-get-exist test

Comment (ikopylov) [reply .github/workflows/integration-tests.yml:215]: But you need to add auth for `bobt` test, because `bobt` works through REST API

File: CHANGELOG.md [hunk 6]
```diff
@@ -4,13 +4,24 @@ Bob versions changelog
...
+
+## [2.1.0-alpha.7] - 2022-11-28
+#### Added
 - Include bobt into the zip archive attached to the release infrastructure (#669)
 - Include bobc into release builds (#569)
 - Added blob-info and index-info features to brt (#356)
 - Support for files, file name patterns, key ranges and a 'exists' subcommand to 'bobc' (#539)
-- Integration tests (#518)
-- Added integration tests for aliens (#642)
-- Added integration tests for bobt (#648)
```
Comment (ikopylov) [inline CHANGELOG.md:13]: Something bad happened here. Revert all this changes and add normal description of this particular PR

File: integration-tests/generate_from_templates.py [hunk 7]
```diff
@@ -101,5 +103,5 @@ def pathified(string):
 with open("Templates/users_template.yml.j2") as users:
     template = Template(users.read())
     with open(os.path.join(path, 'users.bobnet.yaml'), 'w') as f:
-        f.write(template.render())
+        f.write(template.render(auth_user=args.auth_user, auth_password=args.auth_password))
```
Comment (ikopylov) [inline integration-tests/generate_from_templates.py:106]: You should also enable auth in `node.yaml`

Comment (Ziktorin) [reply integration-tests/generate_from_templates.py:106]: Auth is enabled by passing `vars(args)` into `template.render()` for node config. It is the same as explicitly listing every parameter that was parsed by parse_args() function, and as long as there is a parameter with the same name in a template it will be automatically substituded with an according value. And among arguments there is an `auth_type` argument that sets the auth type depending on value in the matrix.

File: .github/workflows/integration-tests.yml [hunk 8]
```diff
@@ -143,6 +143,7 @@ jobs:
       matrix:
         key: [8, 16]
         mode: [normal, random]
+        auth: [None, Basic]
```
Comment (ikopylov) [inline .github/workflows/integration-tests.yml:146]: With this line we will run 8 tests in parallel. But it is enough to run a single test with auth enabled.
Please, use `include` to add one specific configuration: https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#example-adding-configurations

This line will be `auth: [None]` and include should add `(key: 8, mode: normal, auth: Basic)`

File: .github/workflows/integration-tests.yml [hunk 9]
```diff
@@ -283,6 +289,7 @@ jobs:
     strategy:
       matrix:
         key: [8, 16]
+        auth: [None, Basic]
```
Comment (ikopylov) [inline .github/workflows/integration-tests.yml:292]: Same here. Run test with `Basic` auth only on 8 bytes keys

**Comment by ikopylov** [comment]
Merge master to this branch and redirect request to master

File: .github/workflows/integration-tests.yml [hunk 10]
```diff
@@ -32,15 +33,15 @@ on:
         required: true
         default: '0'
       bob_bobt_count:
-        description: 'Amount of iterations for bobt test'
+        description: 'Amount of iterations for chain operation test'
         required: true
         default: '100000'
       bob_bobt_first_index:
-        description: 'Bottom boundary for key in bobt test'
+        description: 'Bottom boundry for chain test'
```
Comment (ikopylov) [inline .github/workflows/integration-tests.yml:40]: And this

File: .github/workflows/integration-tests.yml [hunk 11]
```diff
@@ -32,15 +33,15 @@ on:
...
         required: true
         default: '0'
       bob_bobt_last_index:
-        description: 'Upper boundary for key in bobt test'
+        description: 'Upper boundry for chain test'
```
Comment (ikopylov) [inline .github/workflows/integration-tests.yml:44]: And this one too

File: .github/workflows/integration-tests.yml [hunk 12]
```diff
@@ -229,7 +239,7 @@ jobs:
           INPUT="${{ github.event.inputs.bob_replicas_amount }}"
           echo "BOB_REPLICAS_AMOUNT=${INPUT:-"2"}" >> $GITHUB_ENV
           INPUT="${{ github.event.inputs.bob_payload_size }}"
-          echo "BOB_PAYLOAD_SIZE=${INPUT:-"4096"}" >> $GITHUB_ENV
+          echo "BOB_PAYLOAD_SIZE=${INPUT:-"10240"}" >> $GITHUB_ENV
```
Comment (ikopylov) [inline .github/workflows/integration-tests.yml:242]: Revert this change

File: .github/workflows/integration-tests.yml [hunk 13]
```diff
@@ -263,6 +273,7 @@ jobs:
       - name: List availible images
         run: |
           docker image ls
+      #Also sets BOB_BOBP_ACTIONS_OFFSET as GITHUB_ENV from check_interval parameter (check_interval + 1s)
```
Comment (ikopylov) [inline .github/workflows/integration-tests.yml:276]: Revert this change

File: .github/workflows/integration-tests.yml [hunk 14]
```diff
@@ -301,11 +316,15 @@ jobs:
           INPUT="${{ github.event.inputs.bob_bobt_first_index }}"
           echo "BOB_BOBT_FIRST_INDEX=${INPUT:-"0"}" >> $GITHUB_ENV
           INPUT="${{ github.event.inputs.bob_bobt_last_index }}"
-          echo "BOB_BOBT_LAST_INDEX=${INPUT:-"10000"}" >> $GITHUB_ENV
+          echo "BOB_BOBT_LAST_INDEX=${INPUT:-"100000"}" >> $GITHUB_ENV
```
Comment (ikopylov) [inline .github/workflows/integration-tests.yml:319]: Revert this change

File: .github/workflows/integration-tests.yml [hunk 15]
```diff
@@ -333,15 +352,16 @@ jobs:
       - name: List availible images
         run: |
           docker image ls
+      #Also sets BOB_BOBP_ACTIONS_OFFSET as GITHUB_ENV from check_interval parameter (check_interval + 1s)
```
Comment (ikopylov) [inline .github/workflows/integration-tests.yml:355]: This one too

File: .github/workflows/integration-tests.yml [hunk 16]
```diff
@@ -32,15 +33,15 @@ on:
         required: true
         default: '0'
       bob_bobt_count:
-        description: 'Amount of iterations for bobt test'
+        description: 'Amount of iterations for bobt operation test'
         required: true
         default: '100000'
       bob_bobt_first_index:
-        description: 'Bottom boundary for key in bobt test'
+        description: 'Bottom boundry for key in bobt test'
```
Comment (ikopylov) [inline .github/workflows/integration-tests.yml:40]:
```suggestion
        description: 'Bottom boundary for key in bobt test'
```

File: .github/workflows/integration-tests.yml [hunk 17]
```diff
@@ -32,15 +33,15 @@ on:
...
         required: true
         default: '0'
       bob_bobt_last_index:
-        description: 'Upper boundary for key in bobt test'
+        description: 'Upper boundry for key in bobt test'
```
Comment (ikopylov) [inline .github/workflows/integration-tests.yml:44]:
```suggestion
        description: 'Upper boundary for key in bobt test'
```

File: .github/workflows/integration-tests.yml [hunk 18]
```diff
@@ -1,6 +1,7 @@
 name: integration-tests
 on:
   push:
+    branches: ['672-integrationtests-run-tests-with-and-without-authorization'] #REMOVE WHEN ON MASTER
```
Comment (ikopylov) [inline .github/workflows/integration-tests.yml:4]: Remove this line

**Review by ikopylov — APPROVED** [review]
LGTM
